geode-connectors/src/distributedTest/java/org/apache/geode/connectors/jdbc/internal/cli/ListMappingCommandDUnitTest.java

```diff
@@ -74,31 +81,147 @@ public void listsRegionMappingFromClusterConfiguration() throws Exception {
+        "create jdbc-mapping --region=" + regionName + " --data-source=connection "
+            + "--table=myTable --pdx-name=myPdxClass";
+    gfsh.executeAndAssertThat(mapping).statusIsSuccess();
+    mapping =
+        "create jdbc-mapping --region=" + GROUP1_REGION + " --groups=" + TEST_GROUP1
+            + " --data-source=connection " + "--table=myTable --pdx-name=myPdxClass";
+    gfsh.executeAndAssertThat(mapping).statusIsSuccess();
+    mapping =
+        "create jdbc-mapping --region=" + GROUP2_REGION + " --groups=" + TEST_GROUP2
+            + " --data-source=connection " + "--table=myTable --pdx-name=myPdxClass";
+    gfsh.executeAndAssertThat(mapping).statusIsSuccess();
+    mapping =
+        "create jdbc-mapping --region=" + GROUP1_GROUP2_REGION + " --groups=" + TEST_GROUP1 + ","
+            + TEST_GROUP2 + " --data-source=connection " + "--table=myTable --pdx-name=myPdxClass";
+    gfsh.executeAndAssertThat(mapping).statusIsSuccess();
+
+    CommandStringBuilder csbd = new CommandStringBuilder(
+        "describe jdbc-mapping --region=" + regionName + " --groups=" + TEST_GROUP1);
+    gfsh.executeAndAssertThat(csbd.toString());
+
+    {
+      CommandStringBuilder csb =
+          new CommandStringBuilder(LIST_MAPPING);
+      CommandResultAssert commandResultAssert = gfsh.executeAndAssertThat(csb.toString());
+
+      commandResultAssert.statusIsSuccess();
+      commandResultAssert.tableHasRowCount(LIST_OF_MAPPINGS, 1);
+      commandResultAssert.tableHasColumnOnlyWithValues(LIST_OF_MAPPINGS, regionName);
+    }
+
+    {
+      CommandStringBuilder csb =
+          new CommandStringBuilder(LIST_MAPPING + " --groups=" + TEST_GROUP1);
+      CommandResultAssert commandResultAssert = gfsh.executeAndAssertThat(csb.toString());
+
+      commandResultAssert.statusIsSuccess();
+      commandResultAssert.tableHasRowCount(LIST_OF_MAPPINGS, 2);
+      commandResultAssert
+          .tableHasColumnOnlyWithValues(LIST_OF_MAPPINGS, GROUP1_REGION, GROUP1_GROUP2_REGION);
+    }
+
+    {
+      CommandStringBuilder csb =
+          new CommandStringBuilder(LIST_MAPPING + " --groups=" + TEST_GROUP2);
+      CommandResultAssert commandResultAssert = gfsh.executeAndAssertThat(csb.toString());
+
+      commandResultAssert.statusIsSuccess();
+      commandResultAssert.tableHasRowCount(LIST_OF_MAPPINGS, 2);
+      commandResultAssert
+          .tableHasColumnOnlyWithValues(LIST_OF_MAPPINGS, GROUP2_REGION, GROUP1_GROUP2_REGION);
+    }
+
+    {
+      CommandStringBuilder csb =
+          new CommandStringBuilder(LIST_MAPPING + " --groups=" + TEST_GROUP1 + "," + TEST_GROUP2);
+      CommandResultAssert commandResultAssert = gfsh.executeAndAssertThat(csb.toString());
+      commandResultAssert.statusIsSuccess();
+      // There will be 4 items: testRegion1 for testGroup1, testRegion2 for testGroup2,
+      // group1Group2Region for testGroup1, group1Group2Region for testGroup2
+      commandResultAssert.tableHasRowCount(LIST_OF_MAPPINGS, 4);
```

Comment:
This doesn't quite make sense. We created 3 mapping with various group values. I would expect the list only show the 3 mappings we created. Here there are 4 mappings in the list result. I would consider one of them duplicate.

Reply:
✅

Reply:
✅